### PR TITLE
feat(ui): Add collapsible panels

### PR DESCRIPTION
**Pull request by PixelDust22**
https://user-images.githubusercontent.com/8980455/181135307-773f9e59-26f1-4f4b-982a-d87e410abfcf.mov

Standard collapsible panels. By clicking on the header, the body can retract or expand with an animation.

Supports https://github.com/getsentry/sentry/pull/37087

### CODE REVIEW

**Comment by evanpurkhiser** [comment]
@vuluongj20 can you take a look at this?

**Review by evanpurkhiser — CHANGES_REQUESTED** [review]
gonna wait for @vuluongj20 to look

File: static/app/components/panels/panelHeader.tsx [hunk 1]
```diff
@@ -41,3 +42,21 @@ const PanelHeader = styled('div')<Props>`
...
+  collapsed?: boolean;
+  onClick?: () => void;
+}) {
+  return (
+    <ClickablePanelHeader onClick={props.onClick}>
+      {props.children}
+      <IconChevron direction={props.collapsed ? 'right' : 'down'} />
+    </ClickablePanelHeader>
+  );
+}
```
Comment (vuluongj20) [inline static/app/components/panels/panelHeader.tsx:62]: We should remove the `PanelHeader`'s bottom border when `collapsed` is true
image

File: static/app/components/panels/panelHeader.tsx [hunk 2]
```diff
@@ -41,3 +42,21 @@ const PanelHeader = styled('div')<Props>`
 `;
 
 export default PanelHeader;
+
+const ClickablePanelHeader = styled(PanelHeader)`
+  user-select: none;
```
Comment (vuluongj20) [inline static/app/components/panels/panelHeader.tsx:47]: Why do we want to disable text selection in the header? I can see a use case where the user would wanna select and copy the header text

**Review by vuluongj20 — APPROVED** [review]
Left some small comments, but otherwise this looks good. Nice job!

**Comment by Neo-Zhixing** [comment]
@evanpurkhiser Can you remove the merge block if you're 👍 on this?

File: static/app/components/panels/panelBody.tsx [hunk 3]
```diff
@@ -13,3 +14,20 @@ const PanelBody = styled('div')<BaseProps>`
 `;
 
 export default PanelBody;
+
+export const CollapsePanelBody = styled(motion.div)`
```
Comment (evanpurkhiser) [inline static/app/components/panels/panelBody.tsx:18]: damn. I kind of don't like that this has a default export and a named export

Comment (Neo-Zhixing) [reply static/app/components/panels/panelBody.tsx:18]: We can only have one default export per file, so we either need to move `CollapsePanelBody` elsewhere, or we need to change `PanelBody` to a named export which is going to cause a lot of breakage. Imo having named and default export in the same file is totally fine, the IDE should be able to figure things out for you.

**Comment by doralchan** [comment]
Wait let's not add collapsible panels because we have other content that is collapsible as well (accordion, clippedbox, etc) which are all inconsistent. I'd rather have our other components cleaned up than introduce a new functionality to panel (which are primarily reserved for form and table content) doing something unexpected when its often the main content in the page. This will introduce more problems for us in design systems. I'd be happy to discuss the other components more and use cases for our existing panel.

**Comment by Neo-Zhixing** [comment]
Closed after reconciling with Ryan on the design.

**Pull request closed without merging**